### ISSUE TITLE
Cannot create FakeRedis with positional host & port arguments

### ISSUE THREAD

**Issue by macdjord** [dsoftwareinc/fakeredis-py#36]
**Describe the bug**

In 1.8.1, if you try to create an instance of `FakeRedis` and pass in the the host & port args positionally, rather than by keyword - which is a perfectly valid and functional way of instantiating `Redis` - it will fail:
* `FakeRedisMixin.__init__()` sets any keyword argument which you don't give it to the default value extracted from the signature of `Redis.__init__()` in `**kwds`
* However, it *doesn't* check if any of those arguments have been passed positionally in `*args`
* Thus, when it calls the Redis constructor as `super().__init__(*args, **kwds)`, the `host` argument is set twice - once positionally and once by keyword - which is not legal in Python

**To Reproduce**

This is the code under test:
```python
connection: redis.Redis = redis.Redis(redis_host, redis_port)
```

Patch it to this:
```python
connection: redis.Redis = fakeredis.FakeRedis(redis_host, redis_port)
```

**Expected behavior**

`FakeRedis` should be a drop-in replacement for `Redis`

**Actual behaviour**

```text
/usr/local/lib/python3.7/site-packages/fakeredis/_server.py:145: TypeError: __init__() got multiple values for argument 'host'
```

**Desktop (please complete the following information):**
 - OS: Alpine Linux v3.13
 - Python version: 3.7.10
 - redis-py version: 3.2.1
 - fakeredis version: 1.8.1

**Comment by cunla** [comment]
Hi, can you please post the full test code, I think you are using fakeredis incorrectly.
You should not override `Redis`, but create a connection using `FakeRedis` instead of `Redis`

See [How to use](https://github.com/dsoftwareinc/fakeredis-py#how-to-use) section.

**Comment by macdjord** [comment]
@cunla: I am using FakeRedis for unit tests. Since the code that creates the Redis connection is part of what's being tested, I'm using `unittest.mock.patch()` to replace the call to `redis.Redis()` with `fakeredis.FakeRedis()`.

**Comment by cunla** [comment]
Can you post a sample test that fails here? I suspect (though not sure) you are patching in the wrong place.
Notice `FakeRedis` inherits from `Redis` so it might cause a circular dependency.

**Comment by cunla** [comment]
@macdjord ?

**Comment by macdjord** [comment]
@cunla I can't give you the original code due to company policy. However, it looks something like this:

`foo.py`:
```python
from redis import Redis

def connect_redis(redis_host: str, redis_port: int) -> Redis:
    redis_con = Redis(redis_host, redis_port)
    ...
    return redis_con

def bar(zip, zap):
    ...
    redis_con = connect_redis(redis_host, redis_port)
    ...
```

`test_foo.py`:
```python
from unittest.mock import patch
from fakeredis import FakeRedis 
...
def test_bar():
    # Mock Redis connection
    with patch('foo.Redis', FakeRedis):
        # Call function
        bar(zip, zap)

        # Check the data in Redis is as expected
        ...
```

**Comment by whoiswes** [comment]
I use a very similar pattern (monkeypatch redis.Redis with FakeRedis) and can confirm this bug.

FakeRedis 1.7.1 works fine, anything after is broken.

OS: Debian Bullseye, CentOS 7
Python version: 3.9.2
redis-py version: 4.3.4
fakeredis version: 1.8.1

**Comment by cunla** [comment]
Running this test passed.
I think instead of mocking `foo.Redis` you should mock `redis.Redis`.
```
from redis import Redis


def connect_redis(redis_host: str, redis_port: int) -> Redis:
    redis_con = Redis(redis_host, redis_port)
    return redis_con


def bar(zip, zap):
    redis_con = connect_redis('localhost', 6379)


from unittest.mock import patch
from fakeredis import FakeRedis


def test_bar():
    # Mock Redis connection
    with patch('redis.Redis', FakeRedis):
        # Call function
        bar(zip, [])
```
with

**Comment by cunla** [comment]
Fix in 1.8.2
